Autopilot/Path Manager/PathManager.c

```diff
@@ -124,14 +118,20 @@ void pathManagerInit(void) {
 #endif
 }
 
+char buffer[200];
+
 void pathManagerRuntime(void) {
 #if DEBUG
 //    char str[16];
 //    sprintf(&str,"%f",pmData.time);
 //    UART1_SendString(&str);
 #endif
-    copyGPSData();
+    requestGPSInfo();
 
+    if (isNewGPSDataAvailable()){
+        copyGPSData();
+    }
+    
```

Comment:
This will cause interchip to not update until there's new GPS data. This check should be moved into the `copyGPSData` function, for just the section using GPS data. `copyGPSData` might need to be renamed, too.